python/nano/tutorial/inference/tensorflow/cifar10.py

```diff
@@ -0,0 +1,134 @@
+
+# Scale images to the [0, 1] range
+x_train = x_train.astype("float32") / 255
+x_test = x_test.astype("float32") / 255
+print("x_train shape:", x_train.shape)
+print(x_train.shape[0], "train samples")
+print(x_test.shape[0], "test samples")
+
+
+# convert class vectors to binary class matrices
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+
+model = keras.Sequential()
+
+model.add(layers.Conv2D(32, (3,3), padding='same', activation='relu', input_shape=(32,32,3)))
+model.add(layers.BatchNormalization())
+model.add(layers.Conv2D(32, (3,3), padding='same', activation='relu'))
+model.add(layers.BatchNormalization())
+model.add(layers.MaxPooling2D(pool_size=(2,2)))
+model.add(layers.Dropout(0.3))
+
+model.add(layers.Conv2D(64, (3,3), padding='same', activation='relu'))
+model.add(layers.BatchNormalization())
+model.add(layers.Conv2D(64, (3,3), padding='same', activation='relu'))
+model.add(layers.BatchNormalization())
+model.add(layers.MaxPooling2D(pool_size=(2,2)))
+model.add(layers.Dropout(0.5))
+
+model.add(layers.Conv2D(128, (3,3), padding='same', activation='relu'))
+model.add(layers.BatchNormalization())
+model.add(layers.Conv2D(128, (3,3), padding='same', activation='relu'))
+model.add(layers.BatchNormalization())
+model.add(layers.MaxPooling2D(pool_size=(2,2)))
+model.add(layers.Dropout(0.5))
+
+model.add(layers.Flatten())
+model.add(layers.Dense(128, activation='relu'))
+model.add(layers.BatchNormalization())
+model.add(layers.Dropout(0.5))
+model.add(layers.Dense(num_classes, activation='softmax'))  
+
+# this line is optional
+# model = Model(inputs=model.inputs, outputs=model.outputs)
+
+model.summary()
+
+batch_size = 128
+epochs = 20
+
+model.compile(loss="categorical_crossentropy",
+              optimizer="adam", metrics=["accuracy"])
+
+model.fit(x_train, y_train, batch_size=batch_size,
+          epochs=epochs, validation_split=0.1)
+
+score = model.evaluate(x_test, y_test, verbose=0)
+print("Test loss:", score[0])
+print("Test accuracy:", score[1])
+
+
+# Quantization using Intel Neural Compressor
+# pip install 'neural-compressor'
+
+# calib_dataset only accept tf.data.Dataset object
+tune_dataset = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+
+# Execute quantization
+q_model = model.quantize(calib_dataset=tune_dataset)
+
+# Inference using quantized model
+y_test_hat = q_model(x_test)
+
+# Evaluate the quantized model
+loss = float(tf.reduce_mean(
+             losses.categorical_crossentropy(y_test, y_test_hat)))
+categorical_accuracy = metrics.CategoricalAccuracy()
+categorical_accuracy.update_state(y_test, y_test_hat)
+accuracy = categorical_accuracy.result().numpy()
+
+print("Quantization test loss:", loss)
+print("Quantization test accuracy:", accuracy)
+# Note: accuracy decrease varies from different tasks and situations,
+# and sometimes the accuracy is even higher.
+# You can set a quantization threshold when making a quantization model.
```

Comment:
how long does this example run?

Reply:
About half an hour.

Reply:
a little bit long for a fundamental example yah? 

@MeouSker77 how much time does it take to run through https://github.com/intel-analytics/BigDL/blob/main/python/nano/tutorial/training/tensorflow/tensorflow_train_multi_instance.py

I wonder if we can use this model and data in this example.

No comment on other parts of the code

Reply:
> @MeouSker77 how much time does it take to run through https://github.com/intel-analytics/BigDL/blob/main/python/nano/tutorial/training/tensorflow/tensorflow_train_multi_instance.py

about 7 minutes

Reply:
> I wonder if we can use this model and data in this example.

I remember using the model and data in that example may encounter the conflict of tcmalloc and tf quantization

Reply:
> > @MeouSker77 how much time does it take to run through https://github.com/intel-analytics/BigDL/blob/main/python/nano/tutorial/training/tensorflow/tensorflow_train_multi_instance.py
> 
> about 7 minutes

I uses `num_epochs = int(os.environ.get('NUM_EPOCHS', 10))` to get the epoch num, then `export NUM_EPOCHS=1` in bash script before running that example to accelerate it.

Reply:
In my machine, this example takes about an hour to finish. (10 epochs)
The cifar10 example takes 1.5 minutes for an epoch, and half an hour in total. (20 epochs)